docs/design.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Design
+---
+
+## Architectural Overview
+
+This chapter contains the technical documentation for the design.
+
+The Development Environment Manager (DEM) provides an abstraction layer for the Development 
+Platform, which encompasses a collection of available resources, such as registries, catalogs, 
+container engines, and more. The modules that compose the platform form the core of the DEM.
+
+The DEM implements commands that can interact with the platform. These commands can be executed 
+using the command line interface (CLI). Additionally, some commands utilize a Text-based User 
+Interface (TUI) that can be presented on the user's terminal.
+
+![dem_architecture](wp-content/dem_architecture.png){: .center}
+
+The CLI can communicate with the platform in two ways:
+
+1. Over the Platform API, which can be achieved by instantiating the DevEnvLocalSetup class.
+2. By registering a UserOutput class, which is used by the core modules to interact with the user.
+
+The platform offers the following resources:
+
+1. **Tool images** (`tool_images`): This resource allows you to retrieve both local and registry tool 
+images.
+2. **Container Engine** (`container_engine`): his resource covers various operations related to the 
+container engine, including:
+    - Pulling images
+    - Running containers
+    - Removing images
+    - Searching for images
+3. **Registries** (`registries`): This resource contains information about the available registries.
+    - Adding or deleting registries
+    - Listing the images stored in a registry
+    - Pulling an image from a registry
+4. **Local Development Environments** (`local_dev_envs`):  This resource provides a list of local 
+Development Environments. You can obtain information about a specific Development Environment using 
+this resource.
+5. **Development Environment Catalogs** (`dev_env_catalogs`): This resource deals with the available 
+catalogs. You can perform the following actions:
+    - Adding or deleting catalogs
+    - Listing the Development Environments within a catalog
+    - Cloning a Development Environment descriptor from a catalog 
+
```

Comment:
ADD CTA to XM discord channel : 

{% cta https://discord.com/invite/Nv6hSzXruK) %}  Join our Community:  Everything about Embedded Developer Tools and Developer Platform {% endcta %}

Reply:
please add this to the other pages as well

Reply:
Will be fixed in [DEM-226](https://axem.atlassian.net/browse/DEM-226).